harness/determined/common/experimental/user.py

```diff
@@ -0,0 +1,103 @@
+        agent_uid: Optional[int] = None,
+        agent_gid: Optional[int] = None,
+        agent_user: Optional[str] = None,
+        agent_group: Optional[str] = None,
+    ):
+        self.username = username
+        self.admin = admin
+        self.user_id = user_id
+        self.active = active
+        self.agent_uid = agent_uid
+        self.agent_gid = agent_gid
+        self.agent_user = agent_user
+        self.agent_group = agent_group
+        self.session = session
+        self.display_name = display_name
+
+    def update(
+        self,
+        username: Optional[str] = None,
+        active: Optional[bool] = None,
+        password: Optional[str] = None,
+        agent_uid: Optional[int] = None,
+        agent_gid: Optional[int] = None,
+        agent_user: Optional[str] = None,
+        agent_group: Optional[str] = None,
+        admin: Optional[bool] = None,
+    ) -> bindings.v1PatchUserResponse:
+        v1agent_user_group = bindings.v1AgentUserGroup(
+            agentGid=agent_gid,
+            agentGroup=agent_group,
+            agentUid=agent_uid,
+            agentUser=agent_user,
+        )
+        patch_user = bindings.v1PatchUser(
+            username=username, password=password, active=active, agentUserGroup=v1agent_user_group
+        )
+
+        patch_user_req = bindings.v1PatchUserRequest(userId=self.user_id, user=patch_user)
+        resp = bindings.patch_PatchUser(self.session, body=patch_user_req, userId=self.user_id)
+        return resp
+
+    def rename(self, new_username: str) -> bindings.v1PatchUserResponse:
+        patch_user = bindings.v1PatchUser(username=new_username)
+        patch_user_req = bindings.v1PatchUserRequest(userId=self.user_id, user=patch_user)
+        resp = bindings.patch_PatchUser(self.session, body=patch_user_req, userId=self.user_id)
+        return resp
+
+    def activate(self) -> bindings.v1PatchUserResponse:
+        patch_user = bindings.v1PatchUser(active=True)
+        patch_user_req = bindings.v1PatchUserRequest(userId=self.user_id, user=patch_user)
+        resp = bindings.patch_PatchUser(self.session, body=patch_user_req, userId=self.user_id)
+        return resp
+
+    def deactivate(self) -> bindings.v1PatchUserResponse:
+        patch_user = bindings.v1PatchUser(active=False)
+        patch_user_req = bindings.v1PatchUserRequest(userId=self.user_id, user=patch_user)
+        resp = bindings.patch_PatchUser(self.session, body=patch_user_req, userId=self.user_id)
+        return resp
+
+    def change_display_name(self, display_name: str) -> bindings.v1PatchUserResponse:
+        patch_user = bindings.v1PatchUser(displayName=display_name)
+        patch_user_req = bindings.v1PatchUserRequest(userId=self.user_id, user=patch_user)
+        resp = bindings.patch_PatchUser(self.session, body=patch_user_req, userId=self.user_id)
+        return resp
+
+    def change_password(
+        self, new_password: str, is_hashed: Optional[bool] = False
```

Comment:
let's not expose this `is_hashed` argument here — I don't think a user should bother themselves with this implementation detail. we should always just hash and send it hashed from the client. 
instead having an option, just accept unhashed password and call `api.salt_and_hash` on it here in the sdk, like the CLI is doing at the moment. right now there's a lot of `api.salt_and_hash` calls sprinkled in user cli, and I think this should be "internalized" within SDK / auth code. 

with this change, I wonder if `isHashed` parameter of the `PostUserRequest` and `PatchUserRequest` API requests would still be necessary.

Reply:
As discussed in slack, I'm keeping the isHashed parameter for PostUserRequest and PatchUserRequest. In the CLI and SDK the password will always be hashed before making the API call. But previously the password in PostUserRequest (Even in SetUserPassword API) was assumed to be unhashed. So want to allow that flexibility.